nature/commands/cat.lua

```diff
@@ -17,7 +19,12 @@ usage: cat [file]...]]
 			goto continue
 		end
 
-		sinks.out:writeln(f:read '*a')
+		while true do
+			local block = f:read(chunk_size)
+			if not block then break end
+			sinks.out:write(block)
+		end
+		sinks.out:writeln("")
```

Comment:
```suggestion
```

Reply:
This was actually put in for a reason. When multiple files are being printed, the newline helps determine where one file ends and another begins. That being said, if this is the last file in the loop, then this newline isn't necessary.

Reply:
> If I remember correctly, your original code actually did this, placing a newline after every file (because that's how writeln works).

That is true, but now I realized the extra newline doesn't need to be printed. Checking the `/usr/bin/cat` on my system it does not print an extra newline, which I think is correct.

Reply:
OK. I'm wondering, then, how `cat` behaves when it's fed multiple files that don't end in a newline? We obviously want to mimic that behavior, so if it does indeed behave as you say, then I'll go ahead and revert that.